lib/ruby_smb/dcerpc/winreg.rb

```diff
@@ -194,9 +194,7 @@ def close_key(handle)
       def query_info_key(handle)
         query_info_key_request_packet = RubySMB::Dcerpc::Winreg::QueryInfoKeyRequest.new(hkey: handle)
         query_info_key_request_packet.lp_class = ''
-        query_info_key_request_packet.lp_class.referent.actual_count = 0
-        query_info_key_request_packet.lp_class.maximum_length = 1024
-        query_info_key_request_packet.lp_class.buffer.referent.max_count = 1024 / 2
+        query_info_key_request_packet.lp_class.set_maximum_length(1024)
```

Comment:
Absolutely, I added a `BUFFER_SIZE` constant and updated a few other calls where a an hardcoded value were used. This value was taken from queries captured between Windows client/server. I used `1024` exclusively since it seems to be the common size for buffers. I also updated the `RrpUnicodeString` and `RpcUnicodeString` to improve readability.